File: in_Java/0019 Remove Nth Node From End of List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        int leng = 0;
        ListNode d = head;
        while(d != null){
            d = d.next;
            leng++;
        }
        if(n == leng) return head.next;
        ListNode nh = head;
        int dis = leng - 1 - n;
        while(dis != 0){
            nh = nh.next;
            dis--;
        }
        nh.next = nh.next.next;
        return head;
    }
}

File: in_Python/0019 Remove Nth Node From End of List.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def removeNthFromEnd(self, head, n):
        """
        :type head: ListNode
        :type n: int
        :rtype: ListNode
        """
        h1 = head
        h2 = head
        pre = ListNode(0)
        pre.next = head
        while n != 0:
            h2 = h2.next
            n -= 1
        if h2 == None:
            return pre.next.next
        while h2.next != None:
            h1 = h1.next
            h2 = h2.next
        h1.next = h1.next.next
        return head
        